front/src/applications/stdcmV2/components/StdcmResults/StdcmResults.tsx

```diff
@@ -1,105 +1,130 @@
+  const { t } = useTranslation('stdcm', { keyPrefix: 'simulation.results' });
 
   const [mapCanvas, setMapCanvas] = useState<string>();
 
+  const selectedSimulation = simulationsList[selectedSimulationIndex];
+
   const simulationReportSheetNumber = generateCodeNumber();
 
-  const operationalPointsList = useMemo(
-    () =>
-      getOperationalPointsWithTimes(
-        pathProperties?.suggestedOperationalPoints || [],
-        stdcmData.simulation,
-        stdcmData.departure_time
-      ),
-    [pathProperties, stdcmData]
-  );
+  const isSelectedSimulationRetained = selectedSimulationIndex === retainedSimulationIndex;
+
+  const operationalPointsList = useMemo(() => {
+    if (!selectedSimulation || !selectedSimulation.outputs) {
+      return [];
+    }
+
+    return getOperationalPointsWithTimes(
+      selectedSimulation.outputs.pathProperties?.suggestedOperationalPoints || [],
+      selectedSimulation.outputs.results.simulation,
+      selectedSimulation.outputs.results.departure_time
+    );
+  }, [selectedSimulation]);
 
   return (
     <main className="stdcm-v2-results">
-      <div className="simuation-banner">
-        <div
-          className="simulation-validated"
-          style={{ color: isSimulationSelected ? '#0B723C' : '' }}
-        >
-          {t('simulation.results.simulationNumber')}
-          {isSimulationSelected && (
-            <div className="check-circle">
-              <CheckCircle variant="fill" />
-            </div>
-          )}
+      {showSuccessBanner && (
+        <div className="simulation-available">
+          <span>{t('available')}</span>
         </div>
-        <div className="creation-date">{date}</div>
-      </div>
-      <div className="simuation-results">
-        <div className="results-and-sheet">
-          <StcdmResultsTable
-            stdcmData={stdcmData}
-            operationalPointsList={operationalPointsList}
-            isSimulationSelected={isSimulationSelected}
-            setIsSimulationSelected={setIsSimulationSelected}
-            setInteractedResultsElements={setInteractedResultsElements}
-          />
-          <StdcmUpgrade />
-          {isSimulationSelected && (
-            <div className="get-simulation">
-              <div className="download-simulation">
-                <PDFDownloadLink
-                  document={
-                    <SimulationReportSheetV2
-                      stdcmData={stdcmData}
-                      simulationReportSheetNumber={simulationReportSheetNumber}
-                      mapCanvas={mapCanvas}
-                      operationalPointsList={operationalPointsList}
-                    />
-                  }
-                  fileName={`STDCM-${simulationReportSheetNumber}.pdf`}
-                >
-                  <Button label={t('simulation.results.downloadSimulationSheet')} />
-                </PDFDownloadLink>
+      )}
+
+      <StdcmSimulationBanner
```

Comment:
The button now appears and disappears as intended.

The viewport changes because the configuration form takes up more space in the last simulation compared to the first ones. So for me it's normal.

Reply:
The button now appears and disappears as intended.

The viewport changes because the configuration form takes up more space in the last simulation compared to the first ones. So for me it's normal.

Reply:
Ok got it. But sometimes, when switching between a simulation with few inputs to another with many, we totally lose the viewport of the simulation results which I found weird because if I click on it, I want to see the results without having to scroll again. Do you know what I mean ?

Not sure how to do it, but it would be nice, when we switch, to at least keep the viewport with the begining of the results. The very best would be to keep the viewport (the height position of the navigation where I click according to the whole page) at the exact same spot so we don't see the screen moving but only a glitch at the top if the form/map form have been moving.

Reply:
I haven't found an easy way to do it.

Reply:
We will deal with that in a future PR if necessary.